WordPress/gutenberg
# Enhance link control UI with rich URL previews

![Screen Capture on 2021-06-16 at 15-29-55](https://user-images.githubusercontent.com/444434/122238049-bbcc5680-ceb7-11eb-9867-4c6a274914a1.gif)

## Description

**Please note**: currently the rich previews are only enabled within the post editor and only then within the format library implementation which is utilised by blocks such as `core/paragraph`. This intentional, in order to limit the surface area exposure to the change and allow it to "bed" in. We will enable across other areas of the editor in future subject to it working well where it is currently.

Building on https://github.com/WordPress/gutenberg/pull/18042 and https://github.com/WordPress/gutenberg/pull/31085, this PR wires things together to allow meta data about direct entry URLs (eg: `www.wordpress.org`) to be shown in the `<LinkControl>` UI (the UI that is used to add hyperlinks in Gutenberg).

This currently works in the Post Editor richtext only. It will not work (yet!) in 

* Site Editor
* Nav Editor
* Widget Editor

Closes https://github.com/WordPress/gutenberg/issues/31466

## How has this been tested?




Manual testing and component tests have been completed or added.

To manually test you should:

* Create some text in a paragraph block.
* Create several links (at least 3).
* Click between each link and see if the preview loads.
* Try different URLs and websites.
* Repeat but with your Network speed throttled. Especially try clicking rapidly between several previews to force bugs.

## Screenshots 

https://user-images.githubusercontent.com/444434/121383960-c464e000-c93f-11eb-88e7-b0421d6131cf.mov


## Types of changes





## Checklist:
